stretch_core/stretch_core/stretch_driver.py

```diff
@@ -203,6 +204,19 @@ def command_mobile_base_velocity_and_publish_state(self):
             t.transform.rotation.w = q[3]
             self.tf_broadcaster.sendTransform(t)
 
+            b = TransformStamped()
+            b.header.stamp = current_time
+            b.header.frame_id = self.base_frame_id
+            b.child_frame_id = "base_footprint"
```

Comment:
Why not use "base_link" instead of "base_footprint"?

Reply:
The base_link frame is supposed to be located at the midpoint of wheel centers, whereas, the base_footprint frame is the projection of the base_link frame on the ground plane. Although for Stretch, the base_link frame is analogous to the base_footprint frame, I was wondering if this might change later. 

Added base_footprint because AMCL relies on it, but could as well change the parameter to base_link.